tests/integration_tests/resources/dockerfiles/__init__.py

```diff
@@ -0,0 +1,42 @@
+########
+# Copyright (c) 2018 Cloudify Platform Ltd. All rights reserved
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from io import StringIO
+
+# TODO: Write Ubuntu Wagon Builder Image Dockerfile.
+
+# flake8: noqa
+
+centos_content = """FROM amd64/centos:latest
```

Comment:
We build the image using a method in `cloudify-manager/integration_tests/framework/docker_interface.py`. We are going to expect the average plugin developer to execute tests from the root directory of the plugin. So every plugin developer will need to override the path to the file to be appropriate for their dev environment. This is likely to cause annoyance to other developers, because if someone's dev environment is not set up as mine is they will need to figure out how to type the correct path to the Dockerfile, e.g. `../../cloudify-manager/tests/integration_tests/resources/Dockerfile`. Later, we will want to add testing in the nightlies to run integration tests on some plugins. So we will in that case be executing the tests with cwd `cloudify-manager`.
Also, we may want to have multiple Dockerfile's in the future, for example for Ubuntu VMs and RedHat VMs. In such as case, we need to have a folder structure like this `./../cloudify-manager/tests/integration_tests/resources/{Ubuntu,Centos,RedHat}/`, each with a Dockerfile. It seems very messy to me.
We could require the Dockerfile to be in the root directory of the plugin, and that is an option, but then, since most plugins require nothing more unique than that provided in the Dockerfile, it's just a bunch of duplication.
Since the Python library allows us to pass a fileobj, this means that we never need to fool around with setting the correct path, unless the user needs to override the docker build method to use a path to his or her own Dockerfile. In which case, it is very easy also to override the docker build method to use a file path instead of a file object.

Reply:
How do you imagine a plugin developer will use this framework? just by installing that package or the whole manager? 

if only the framework, so it will be in a fixed relative location. if not, I can see your point.
Also is there are reason not to upload this docker image to docker-hub?